src/Iris/Env.hs

```diff
@@ -144,6 +152,15 @@ newtype CliEnvException = CliEnvException
 
 {- |
 
+@since 0.0.0.0
+-}
+
+data ParserOptions (cmd :: Type) = ParserOptions
+    { cmdOption :: cmd
+    , interactiveOption :: InteractiveMode
+    }
```

Comment:
Let's change the name of the data type and use the data type name as prefix of fields to avoid ambiguity. `ParserOptions` implies "options for a parser" while it's not actually options, it's a result of a given parser, decorated by some additional parsers. Let's just go with a simple `Cmd` type here:

```suggestion
data Cmd (cmd :: Type) = Cmd
    { cmdInteractiveMode :: InteractiveMode
    , cmdCmd :: cmd
    }
```